### ISSUE TITLE
DELETE /ob/listing API not working

### ISSUE THREAD

**Issue by christroutner** [OpenBazaar/openbazaar-go#766]
I've been testing the `/ob/listing` API as per [the API documentation](https://api.openbazaar.org). I got the create (`POST`) and update (`PUT`) versions of the API to work just fine. However, `DELETE` is responding with `404 - {"success":false,"reason":"Listing not found."}`.

I strongly suspect this is a bug, because the inputs for `PUT` and `DELETE` are the same: the listing slug, and `PUT` is working.

**Comment by christroutner** [comment]
If it's possible to prioritize this fix, I'd sure appreciate it. I'm on the cusp of being able to hook [P2P VPS](http://p2pvps.org) to the OB network, but it's going to be weird if people can't remove their listings.

**Comment by cpacia** [comment]
It's expecting DELETE /ob/listing/<slug> what are you entering?

**Comment by christroutner** [comment]
Yes, I'm sending DELETE, but perhaps I'm misunderstanding the [OB API reference](https://api.openbazaar.org/#b14af9a6-8686-5e8b-c92a-3fd8d9f0510e).

[Here is the prototype code that I'm executing against the API](https://github.com/RPiOVN/p2pvps-server/blob/39785fdafd4dff17c24104ce9ae35c4d46ef07d5/site-template-connextcms/keystone/routes/api/openbazaar.js). It's Node.js JavaScript, and I'm using the [request-promise library](https://github.com/request/request-promise).

If you compare the `updateListing()` and `removeMarketListing()` functions, you can see the first is sending PUT and the second is sending DELETE. 

One thing that confused me about the API documentation is that is says in the 'Update the Listing' section: *The currentSlug field is used specify the listing being updated. If the listing contains a new slug, the listing will be renamed using the new slug.* 

I assume that means there is a new key called `currentSlug`, in addition to the normal listing data. But there is no key of that name in the example code to the right. Regardless, I didn't add the `currentSlug` key, but I was still able to update the listing. (But didn't play around with changing the slug.)

I'm following the code example to the right of the 'Delete the Listing' section, but I'm getting the failure.

**Comment by christroutner** [comment]
There is a chance I could be creating a malformed input value. If you don't think it's a bug, I'll play around with it the code and see what I can do about inspecting the headers. Node headers are a little harder to debug, but I can try to recreate the issue with jQuery.

**Comment by christroutner** [comment]
Ok. No bug here. I figured out what I was doing wrong.

I misinterpreted the API docs. Instead of using `PUT /ob/listing` and specifying the changes in the data, I needed to send `PUT /ob/listing/<slug>`. This isn't quite so strict for the PUT updates, as it will update the listing weather the slug is specified in the URI or not. However, DELETE is much pickier.

My problem is I was doing `DELETE /ob/listing` and then putting the `{slug: <slug>}` in the attached data. I need to instead execute `DELETE /ob/listing/<slug>`.
